processes/code-styles/swift/swift-style-guide-en.md

```diff
@@ -1,59 +1,269 @@
+    + [5.9 Error Handling **TODO**](#59-error-handling-todo)
+    + [5.10 Using `guard` Statements](#510-using-guard-statements)
+    + [5.11 Classes and Structures. Which one to use?](#511-classes-and-structures-which-one-to-use)
+    + [5.12 Types](#512-types)
+    + [5.13 Struct Initializers](#513-struct-initializers)
+  * [5.14 Functions vs Methods](#514-functions-vs-methods)
+  * [5.15 Selectors](#515-selectors)
+  * [5.16 Correctness](#516-correctness)
+  * [6. Documentation and Comments](#6-documentation-and-comments)
+    + [6.1 Documentation](#61-documentation)
+    + [6.2 Comments](#62-comments)
+  * [7. Credits](#7-credits)
+
+## 1. File structure
+
+* **2.11** Filenames should be named after type they contain. Don't use abbreviations.
+
+**ConnectionTableViewCell.swift**
+
+```swift
+class ConnectionTableViewCell: UITableViewCell {
+}
+```
+
+**MyOwnProtocol.swift**
+
+```swift
+protocol MyOwnProtocol {
+}
+```
+
+* **2.12** File folder structure should reflect XCode project folder structure. XCode 9 will do it for you by default when adding new files to a project.
+
+
+
+## 2. Code Formatting
+
+* **2.11** Use 4 spaces for tabs.
+* **2.12** Avoid uncomfortably long lines with a hard maximum of 160 characters per line (Xcode->Preferences->Text Editing->Page guide at column: 160 is helpful for this)
+* **2.13** Ensure that there is a newline at the end of every file.
+* **2.14** Ensure that there is no trailing whitespace anywhere (Xcode->Preferences->Text Editing->Automatically trim trailing whitespace + Including whitespace-only lines).
+* **2.15** Do not place opening braces on new lines
+
+```swift
+class SomeClass {
+    func someMethod() {
+        if x == y {
+            /* ... */
+        } else if x == z {
+            /* ... */
+        } else {
+            /* ... */
+        }
+    }
+
+    /* ... */
+}
+```
+
+* **2.16** Colons **MUST** have no space on the left and one space on the right. Exceptions are the ternary operator `? :` and empty dictionary `[:]`.
+
+**Preferred:**
+
+```swift
+// specifying type
+let pirateViewController: PirateViewController
+
+// dictionary syntax (note that we left-align as opposed to aligning colons)
+let ninjaDictionary: [String: AnyObject] = [
+    "fightLikeDairyFarmer": false,
+    "disgusting": true
+]
+
+// declaring a function
+func myFunction<T, U: SomeProtocol>(firstArgument: U, secondArgument: T) where T.RelatedType == U {
+    /* ... */
+}
+
+// calling a function
+someFunction(someArgument: "Kitten")
+
+// superclasses
+class PirateViewController: UIViewController {
+    /* ... */
+}
+
+// protocols
+extension PirateViewController: UITableViewDataSource {
+    /* ... */
+}
+```
+
+```swift
+class TestDatabase: Database {
+var data: [String: CGFloat] = ["A": 1.2, "B": 3.2]
+}
+```
+
+**Not Preferred:**
+
+```swift
+class TestDatabase : Database {
+    var data :[String:CGFloat] = ["A" : 1.2, "B":3.2]
+}
+```
+
+* **2.17** In general, there should be a space following a comma.
```

Comment:
I don't understand. Where should be a space? IMO this phrase is unclear.

Reply:
After each comma there should be a space. Excluding strings.